rust/src/buffer.rs

```diff
@@ -176,22 +176,44 @@ impl MutableBuffer {
         self
     }
 
+    /// Manually sets the `len` of this buffer
+    pub fn set_len(mut self, new_len: usize) -> Self {
+        assert!(
+            new_len <= self.capacity,
+            "Unable to set len outside capacity"
+        );
+        self.len = new_len;
+        self
+    }
+
+    /// Ensure that `count` bytes from `start` contain zero bits
+    ///
+    /// This is used to initialize the bits in a buffer, however, it has no impact on the `len`
+    /// of the buffer and so can be used to initialize the memory region from `len` to `capacity`.
+    pub fn ensure_null_bits(&mut self, start: usize, count: usize) {
```

Comment:
Instead of having two extra methods, can we add a single method (maybe replace the existing `with_bitset`) that:
1. fill the range between `len` and `capacity` with 0s.
2. increase the `len` to be the same as current `capacity`.

Then we can just call this method either after creating a new buffer or resizing existing buffer.

Reply:
>     1. fill the range between `len` and `capacity` with 0s.
This is what `ensure_null_bits` is used for. 
>     2. increase the `len` to be the same as current `capacity`.
However, it does not change the `len`.  I don't think we want to do this until we build when we know what `len` should be.  If `len` is set to capacity the resulting `Buffer`'s `len` would be too big.

I also don't love having the `set_len` method but without it we will have to do alot of checks to see when `len` has to be incremented, i.e. on every `push`

Reply:
Thanks for the explanation. Yes I agree we need two functions. Nit: can we rename `ensure_null_bits` to something like `set_null_bits` to make it more clear (`ensure_null_bits` seems like a check on the validity of the array rather than actually changing it).